content/articles/build-autocorrect-feauture-using-nlp-with-python/index.md

```diff
@@ -0,0 +1,285 @@
+    return probs
+```
+#### Implement 4 edit word functions
+Each of the 4 edit functions is implemented as shown below each performing different tasks as early illustrated. We can also combine the functions to be able to find a list of all the strings that are n edits to perform.
+
+```python
+
+# Now we implement 4 edit word functions
+# delete_letter: given a word, it returns all the possible strings that have one character removed.
+# switch_letter: given a word, it returns all the possible strings that have two adjacent letters switched.
+# replace_letter: given a word, it returns all the possible strings that have one character replaced by another different letter.
+# insert_letter: given a word, it returns all the possible strings that have an additional character inserted.
+
+def delete_letter(word):
+    delete_l = []
+    split_l = []
+    for i in range(len(word)):
+        split_l.append((word[0:i], word[i:]))
+    for a, b in split_l:
+        delete_l.append(a + b[1:])
+    return delete_l
+
+
+delete_word_l = delete_letter(word="cans")
+
+
+def switch_letter(word):
+    split_l = []
+    switch_l = []
+    for i in range(len(word)):
+        split_l.append((word[0:i], word[i:]))
+    switch_l = [a + b[1] + b[0] + b[2:] for a, b in split_l if len(b) >= 2]
+    return switch_l
+
+
+switch_word_l = switch_letter(word="eta")
+
+
+def replace_letter(word):
+    split_l = []
+    replace_l = []
+    for i in range(len(word)):
+        split_l.append((word[0:i], word[i:]))
+    letters = 'abcdefghijklmnopqrstuvwxyz'
+    replace_l = [a + l + (b[1:] if len(b) > 1 else '') for a, b in split_l if b for l in letters]
+    return replace_l
+
+
+replace_l = replace_letter(word='can')
+
+
+def insert_letter(word):
+    split_l = []
+    insert_l = []
+    for i in range(len(word) + 1):
+        split_l.append((word[0:i], word[i:]))
+    letters = 'abcdefghijklmnopqrstuvwxyz'
+    insert_l = [a + l + b for a, b in split_l for l in letters]
+    # print(split_l)
+    return insert_l
+
+
+
+# combining the edits
+# switch operation optional
+def edit_one_letter(word, allow_switches=True):
+    edit_one_set = set()
+    edit_one_set.update(delete_letter(word))
+    if allow_switches:
+        edit_one_set.update(switch_letter(word))
+    edit_one_set.update(replace_letter(word))
+    edit_one_set.update(insert_letter(word))
+    return edit_one_set
+
+# edit two letters
+def edit_two_letters(word, allow_switches=True):
+    edit_two_set = set()
+    edit_one = edit_one_letter(word, allow_switches=allow_switches)
+    for w in edit_one:
+        if w:
+            edit_two = edit_one_letter(w, allow_switches=allow_switches)
+            edit_two_set.update(edit_two)
+    return edit_two_set
+```
+#### Finding Similar/Corrected Word
+The program prompts the user to enter a word, if the word exists in the dictionary it is assumed to be right else a similar word is produced by the program. This is implemented by this piece of code.
```

Comment:
Change to "The program prompts the user to enter a word. If the word exists in the dictionary, it is assumed to be right, or else the program produces a similar word. This piece of code implements this."